lib/heartcheck.rb

```diff
@@ -95,6 +99,13 @@ def info_checks
       checks.select { |ctx| ctx.respond_to?(:info) }
     end
 
+    # an executor class that respond to execute(checkers)
+    #
+    # @return [Heartcheck::Services::ResponseFormatter] or nil
+    def formatter
+      @formatter ||= Heartcheck::Services::ResponseFormatter.new(hash_formatter)
```

Comment:
I would still prefer a default formatter that just returns it's argument. And when setup asks for a hash formatter, that instance gets replaced by the hash formatter version.